### PR TITLE
Clean unused code in pkg/securitycontext/util.go

### LINKED ISSUE #73712: Dead code in pkg/securitycontext/util.go
**What happened**:
I was working on another feature and found that `ParseSELinuxOptions` in `pkg/securitycontext/util.go` code is not called anywhere: https://github.com/kubernetes/kubernetes/blob/2d7b92ee743de20d17406003e463a829a0db5a51/pkg/securitycontext/util.go#L55

A search seems to verify:

https://github.com/kubernetes/kubernetes/search?q=ParseSELinuxOptions&unscoped_q=ParseSELinuxOptions
https://cs.k8s.io/?q=ParseSELinuxOptions&i=nope&files=&repos=

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by WanLinghao**
/kind cleanup

**What this PR does / why we need it**:
Clean unused code in pkg/securitycontext/util.go

Fixes #73712

```release-note
NONE
```

### CODE REVIEW

File: pkg/securitycontext/util.go [hunk 1]
```diff
@@ -47,26 +44,6 @@ func HasCapabilitiesRequest(container *v1.Container) bool {
 	return len(container.SecurityContext.Capabilities.Add) > 0 || len(container.SecurityContext.Capabilities.Drop) > 0
 }
 
-const expectedSELinuxFields = 4
-
-// ParseSELinuxOptions parses a string containing a full SELinux context
-// (user, role, type, and level) into an SELinuxOptions object.  If the
-// context is malformed, an error is returned.
-func ParseSELinuxOptions(context string) (*v1.SELinuxOptions, error) {
```
Comment (php-coder) [inline pkg/securitycontext/util.go]: Given that it's a public method, could it be a breaking change?

Comment (mourya007) [reply pkg/securitycontext/util.go]: Just a query regarding this, How do we parse SELinux context roles, user etc after that ? Do we have other usecase for this ?

**Comment by mourya007** [comment]
cc @liggitt

File: pkg/securitycontext/util.go (hunk 1 above)
Comment (liggitt) [reply pkg/securitycontext/util.go]: > Given that it's a public method, could it be a breaking change?

we do not advertise kubernetes/kubernetes as a utility library. if someone is depending on this, it's a simple enough function to copy into their code

**Pull request merged**